toolbar.py

```diff
@@ -364,10 +366,42 @@ def __init__(self, abiword_canvas):
         self.insert(tool_item, -1)
         tool_item.show()
 
+        separator = Gtk.SeparatorToolItem()
+        separator.set_expand(True)
+        separator.props.draw = False
+        self.insert(separator, -1)
+        separator.show()
+
+        self._word_count_label = Gtk.Label(_("No words"))
+        self._word_count_label.show()
+        tool_item = Gtk.ToolItem()
+        tool_item.add(self._word_count_label)
+        self.insert(tool_item, -1)
+        tool_item.show()
+
         self._abiword_canvas.connect("page-count", self._page_count_cb)
         self._abiword_canvas.connect("current-page", self._current_page_cb)
         self._abiword_canvas.connect("zoom", self._zoom_cb)
 
+        # For some reason, if we don't connect changed event after
+        # the abi canvas is loaded, the activity fails.
+        def internal_callback():
+            self._abiword_canvas.connect_after("changed", self._word_count_cb)
+            self._word_count_cb(self._abiword_canvas)
+
+        GObject.timeout_add(2000, internal_callback)
+
+    def _word_count_cb(self, abiword_canvas):
+        text = abiword_canvas.get_content("text/plain", None)[0]
+        # Find all non-whitespace patterns.
+        word_list = re.findall("(\S+)", text)
+        words_count = len(word_list)
+        label_text = _("No words")
+        if words_count > 0:
+            label_text = _("%s words") % str(words_count)
```

Comment:
Please use ngettext (just Google python ngettext).  Some languages have different plural forms other than 1 and more than 1.